### ISSUE TITLE
分页器el-pagination选择pagesize的时候，弹出的选择框可以调整弹出方向吗？

### ISSUE THREAD

**Issue by vipsunwei** [element-plus/element-plus#805]
分页器选择pagesize的时候，弹出的选择框可以调整弹出方向吗？
向下弹出来超出了可视范围了，
![image](https://user-images.githubusercontent.com/15810277/100988517-c9109c00-358a-11eb-9012-116bd0ca496b.png)

**Comment by JeremyWuuuuu** [comment]
Reasonable, but could you give some demo to reproduce it?

**Comment by vipsunwei** [comment]
> Reasonable, but could you give some demo to reproduce it?

图片传不上来了呀，GitHub不显示图片了又。

大概意思就是本身我的浏览器可视区域高度769px，分页器在页面最下方，选择pagesize的时候，弹出来的选择框在分页器整体下方，就给页面高度撑高了，露出来下方一片白背景，和我的主题风格不一样颜色。
希望可以感知分页器在最下方的时候  ， 向上弹出选择框。

**Comment by SimonaliaChen** [comment]
> > Reasonable, but could you give some demo to reproduce it?
> 
> 图片传不上来了呀，GitHub不显示图片了又。
> 
> 大概意思就是本身我的浏览器可视区域高度769px，分页器在页面最下方，选择pagesize的时候，弹出来的选择框在分页器整体下方，就给页面高度撑高了，露出来下方一片白背景，和我的主题风格不一样颜色。
> 希望可以感知分页器在最下方的时候 ， 向上弹出选择框。

Sounds werid. The `popper` will auto flip when there is no enough room in viewport and should not expand the page. Could you provide a reproduce demo to help us in troubleshooting?

**Comment by JeremyWuuuuu** [comment]
> > Reasonable, but could you give some demo to reproduce it?
> 
> 图片传不上来了呀，GitHub不显示图片了又。
> 
> 大概意思就是本身我的浏览器可视区域高度769px，分页器在页面最下方，选择pagesize的时候，弹出来的选择框在分页器整体下方，就给页面高度撑高了，露出来下方一片白背景，和我的主题风格不一样颜色。
> 希望可以感知分页器在最下方的时候 ， 向上弹出选择框。

I think I might know the root cause of this issue. Will be fixed soon.

**Comment by JeremyWuuuuu** [comment]
@vipsunwei Could you paste the demo code here. Thank you.

**Comment by JeremyWuuuuu** [comment]
need further info to investigation, it looks good on my side

**Comment by vipsunwei** [comment]
@JeremyWuuuuu   我重新创建了一个demo，那个分页器感知在最下方弹出是向上的，但是在我的项目中就是向下的，我对照了两个项目不同的地方，demo中我用的主题是
`plugins: [
    [
      "component",
      {
        libraryName: "element-plus",
        styleLibraryName: "theme-chalk",
      },
    ],
  ],`

我在我的项目中使用了自定义主题，
`plugins: [
    [
      "component",
      {
        libraryName: "element-plus",
        styleLibraryName: "~src/theme",
      },
    ],
  ],`
是官方文档提供的那个，自己切换颜色，然后下载下来放到自己的项目中的，

我把styleLibraryName换回theme-chalk 就能正常感知在最下方 ，然后向上弹出，换到自定义的主题就向下弹了。

补充一点：下载下来的主题文件是这样的  aside.css  button.css  等等，  而项目说找不到el-aside.css  el-button.css 等等 ，我手动一个一个的在样式文件前加了el-前缀。

通过对比我觉得是自定义主题的问题。你们可以试试。

**Comment by vipsunwei** [comment]
@JeremyWuuuuu  
这是我用的版本
`"dependencies": {
    "core-js": "^3.6.5",
    "element-plus": "^1.0.1-beta.3",
    "vue": "^3.0.0",
    "vue-router": "^4.0.0-0"
  },
  "devDependencies": {
    "@vue/cli-plugin-babel": "~4.5.0",
    "@vue/cli-plugin-router": "~4.5.0",
    "@vue/cli-service": "~4.5.0",
    "@vue/compiler-sfc": "^3.0.0",
    "babel-plugin-component": "^1.1.1",
    "sass": "^1.26.5",
    "sass-loader": "^8.0.2"
  }`

**Comment by JeremyWuuuuu** [comment]
@vipsunwei Hey, we are glad that you opened this issue, but unless you provide us a repo to reproduce it, we have no idea what is going on with descriptions, we need to see what was really happening.
The doctors can do nothing without seeing the patient, right? (医生不看到病人本身也没办法做任何事情, 不是吗?)

**Comment by vipsunwei** [comment]
@JeremyWuuuuu 我弄了demo了，https://github.com/vipsunwei/pagination-demo

**Comment by JeremyWuuuuu** [comment]
Thank you, I will verify if this is the bug from `ElementPlus`

**Comment by varlinux** [comment]
@vipsunwei I had a temporary solution to resolve your problem whitch just need to rewrite `.el-select-dropdown`
```
/deep/ .el-select-dropdown {
    position: absolute;
    z-index: 1001;
    border: 1px solid #E4E7ED;
    border-radius: 4px;
    background-color: #FFF;
    box-shadow: 0 2px 12px 0 rgba(0,0,0,.1);
    box-sizing: border-box;
    margin: 5px 0;
    top: -200px; // This fill your scrollbar minus height
}
```

**Comment by varlinux** [comment]
Hi, I think I've found the problem. Actually this is not a bug, I'm ranning the developer's project, there are two reasons why the paging `popper` is not displayed at the top.
1. The hoster using the on-demand loading plug-in. In fact, the `El-Pagination` component encapsulates and using the `El-Popper` component, but the hoster does not import it;
In your `element-plus-ui.js`:
```
import {
  ElPagination,
  ElTable,
  ElTableColumn,
  ElContainer,
  ElMain,
  ElAside,
  ElMenu,
  ElMenuItem,
  ElLoading,
  ElSelect,
  ElOption,
  ElDatePicker,
  ElPopper, // add this
} from "element-plus";

const components = [
  ElPagination,
  ElTable,
  ElTableColumn,
  ElContainer,
  ElMain,
  ElAside,
  ElMenu,
  ElMenuItem,
  ElLoading,
  ElSelect,
  ElOption,
  ElDatePicker,
  ElPopper, // add this
];
```
2. The `Babel.config.js` reference is the index.css style in the custom `theme` folder.

old `Babel.config.js`:
```
module.exports = {
  presets: ["@vue/cli-plugin-babel/preset"],
  plugins: [
    [
      "component",
      {
        libraryName: "element-plus",
        styleLibraryName: "~src/theme",
      },
    ],
  ],
};
```
new `Babel.config.js`:
```
module.exports = {
  presets: ["@vue/cli-plugin-babel/preset"],
  plugins: [
    [
      "component",
      {
        libraryName: "element-plus",
        styleLibraryName: "~node_modules/element-plus/lib/theme-chalk",
      },
    ],
  ],
};
```
And I suspect that the `index.css` in the `theme` folder doesn't belong to the native version of Element-Plus.

**Comment by varlinux** [comment]
This is supposed to be a problem with custom styles, which I solved in the demo by assigning the width and height to el-popper.
Before changing the height [interlinkage](http://chuantu.xyz/t6/741/1614326774x1700338588.png)
After adding the width and height [interlinkage](http://chuantu.xyz/t6/741/1614326693x1033348314.png)

[Complete the operation](http://chuantu.xyz/t6/741/1614327717x1700338641.gif)

By introducing the correct style, I found that the body style was missing from the custom style. It was the body that did not define the width and height that caused the loss of the `El-Popper` width and height

In fact, i think this issure It's the same thing as [#1018 ]

> TIP: I modify `el-popper` default position in source code

**Comment by kooriookami** [comment]
fixed in #2267
